cypress/integration/unit/helpers/shift.helper.spec.ts

```diff
@@ -0,0 +1,51 @@
+/// <reference types="cypress" />
+import { DataRow } from "../../../../src/logic/schedule-logic/data-row";
+import { ShiftHelper } from "../../../../src/helpers/shifts.helper"
+import { ShiftCode, ShiftInfoModel } from "../../../../src/common-models/shift-info.model";
+
+type TestCase = { arr: ShiftInfoModel; exp: Array<number> };
+
+const testData1: ShiftInfoModel = {
+    "0": ["R","DN","W"].map(d => ShiftCode[d]),
+    "1": ["P","D","W"].map(d => ShiftCode[d]),
+    "2": ["D","W","W"].map(d => ShiftCode[d])
+    };
+
+const testData2: ShiftInfoModel = {
+    "0": ["W","W","W"].map(d => ShiftCode[d]),
+    "1": ["W","W","W"].map(d => ShiftCode[d]),
+    "2": ["W","W","W"].map(d => ShiftCode[d])
+    }   
+    
+const testData3: ShiftInfoModel = {
+    "0": ["R","R","R"].map(d => ShiftCode[d]),
+    "1": ["R","R","R"].map(d => ShiftCode[d]),
+    "2": ["R","R","R"].map(d => ShiftCode[d])
+    }   
+    
+const testCases: TestCase[] = [
+    {
+        arr: testData1,
+        exp: [3,2,0],
```

Comment:
Dlaczego [3,2,0]?
D jest zmianą pracującą

Reply:
Dobra, to są wartości dla kolumn

Reply:
["R","DN","W"]
["P","D",  "W"]
["D","W", "W"]
"R", "P", "D" -- są zmianę pracujące, tutaj mamy 3
"D", "DN"     -- również są pracujące, a "W" nie jest, tutaj mamy 2
W ostatniej kolumnie mamy same "W" i tutaj ,mamy 0
Dlatego mamy [3,2,0], jak ja to widzę :)